### DISCUSSION TITLE
Trello to GitHub migration

### DISCUSSION THREAD

**Discussion by riongull** [dashtesting/trello-github-etl#38] [General]
Hello Dash Incubator Admins, 

Let's use this place to discuss feedback related to the Trello to GitHub migration.  This was created primarily to facilitate prod task 9 of the [Incubator on GitHub](https://trello.com/c/m8w82CvE/184-incubator-on-github) migration project on Trello, but we may use it for other tasks as well.

### Prod Task 9 from Trello (1 DASH)

- [ ] review the [migration3](https://github.com/orgs/dashtesting/projects/4) project board and its current views
- [ ] create one or more custom views that interests you
- [ ] report any bugs and request any additional features
- [ ] give general feedback

When you've completed your review, comment here with overall feedback including bug reports and feature requests (which you could make as separate issues and link to them if you'd like).

@riongull I can edit your discussion, is that on purpose?

**Comment by SamKirby22** [comment]
Hi Rion, will drop my observations one by one here

**Reply by riongull** [reply]
Great.  Feel free to edit this comment with other general feedback, and/or a summary of your other comments to use this comment as your claim url.  Which view did you create?  Can you give it a name like "Sam's view" or something so I can see an example of how you'd like to view your data?

**Reply by SamKirby22** [reply]
Gret thanks, will do.

**Reply by SamKirby22** [reply]
Is there a way to create a link to an individual bounty? I can’t find a Share option and when open, the URL retains a link to the view itself not the Bounty.

**Reply by SamKirby22** [reply]
I've gone over this in some more detail and from a functional point of view it looks great. 

Created a view
Renamed a view
Toggled configuration settings
Played with Sort and Filter features
Saved as a New View 
Deleted a New View
Duplicated a View

I didn't touch anything to do with Delete or Archive, as it's not immediately clear how to recover afterwards. I did notice some minor Body Text formatting observations when creating a Bounty before realising this is likely out of scope for now.

I think in general it looks really promising so far. Is there/will there be some kind of QA Strategy for this implementation? Essentially i've only looked at this in terms of very light exploratory testing - not really tried to break it. I'm keen to make sure something of this importance is given proper scrutiny - I have a highly skilled Software Tester if you need one.

Also keen to understand it's governance structure and how this will be kept as centralised as possible - no single point of failure.

**Reply by SamKirby22** [reply]
I just noticed the 'request new features' part of the task - any chance of a Body Text Spellcheck? Oh and Underline (Ctrl U takes me into code)

**Reply by riongull** [reply]
I think you meant "how this will be kept as *de*centralized as possible", right?  
That's a question of permissions, which I haven't really worked out yet.  This also depends on how we structure things.  For example it's possible for us to do this where each admin get's their own organization even, though that might be taking it a little too extreme.  The idea is that we can make it as decentralized as we want, just need to weigh pros and cons.

We won't have control over formatting.  Most text boxes support markdown formatting (which doesn't include underlines, thought **bold** and *italic* are supported).

**Reply by riongull** [reply]
> Is there a way to create a link to an individual bounty? I can’t find a Share option and when open, the URL retains a link to the view itself not the Bounty.

I'm not sure I understand the question.  Each bounty (card on Trello) is its own issue here in GitHub (which have their own link/url).  They open in a new tab when you click on them from the project view.

**Reply by SamKirby22** [reply]
I guess so - do we have a scope list that shows this with other items not yet thought out? I'm keen to see where we are against plan - what's left to deliver, what's left to scope etc etc.

**Reply by riongull** [reply]
The purpose of this task is to help define what's left in the scope. If there are bugs, or missing features, or some admin suggests a better way to organize tasks, bounties, projects, etc then that will expand the scope of this migration work. If not, then we'd be close to being done. Does that answer your question?

**Comment by SamKirby22** [comment]
In general i'm finding the board responsiveness alot slower than Trello. Up to 6 seconds when selecting the Sort feature for example

**Reply by riongull** [reply]
I haven't been able to reproduce this on my 2014 era MacBook Air.  

I tried sorting several different columns on the `Main` view and it responded within 1 second each time.  The main performance issue I'm seeing is it's slow to load more data as I scroll on a large list (like the `Main` view).

Anyone else having performance/responsiveness issues please post them here.

**Reply by wizlee** [reply]
Same observation on scrolling here. Filtering feels ok, however there seems to be more delay when scrolling in table or board layout. 

I am using an old surface pro 3 that is not that powerful to begin with though.

**Comment by SamKirby22** [comment]
Can you explain the Trello Type column and values?

**Reply by riongull** [reply]
Both bounties (cards) and tasks (checkboxes on a card) are imported as GitHub "issues" (and are brought onto the project board).  We made a column for them to distinguish the two different issue types for grouping/sorting/filtering purposes.

**Reply by SamKirby22** [reply]
Ok understood. So this view shows all bounties AND tasks within those bounties in a list format - allowing any member to filter based on what's assigned to them and what they're responsible for - or whatever they wan't. Just getting this clear in my head :-)

**Comment by SamKirby22** [comment]
Are the Status values fixed or variable? Should I be checking these too? If so, can you provide definitions for how they'd relate to our current statuses. We seem to be missing a few and have a few that aren't immediately clear.

**Reply by riongull** [reply]
They are fixed for the migration, but after that they will change, either manually, or automatically based on however we want to define that (which would be my preference).

Status for the migration is defined here:
https://github.com/dashtesting/trello-github-etl/blob/main/example.env#L10

It was basically a renaming of Trello board columns, since some columns represent a real "status" (specification, completed, etc), and some represent more of a "type" (jobs, programmes, etc)

**Reply by SamKirby22** [reply]
Ok great thanks.

**Comment by SamKirby22** [comment]
What's the purpose of the Assignees field? - especially related directly to Trello Type of 'Cards'. How would this differ to the Owner (which I assume holds the name of the Primary Admin?) field?

**Reply by riongull** [reply]
Assignee is a built-in GitHub feature of issues.  Right now we're not really using that field since not all admins are on this GitHub org to assign issues to them.  So instead we made a "Trello Assignee" column that brings in the person assigned to the various Trello tasks.

**Reply by SamKirby22** [reply]
Ah ok, so this Dashboard works differently to Trello by way of hosting Tasks in addition to Bounties hence assignee required for Tasks only. Has there been thought around the use of this field for bounties:

Prim Admin or 
Community member who created the Concept or 
Product/Project Owner/Strategist - if different from Admin or other.

**Reply by riongull** [reply]
Those are good ideas to discuss.  it won't be hard to populate once we've decided what we want.

**Comment by dashameter** [comment]
Tried to create a view -> Was missing permissions
Created a view
Unable to change assignees (It's a a feature) 
I can edit other's comments in this discussion .. (Is it a feature?)

Looking good, I think the tighter github integration will be good, until we replace github with something more decentralized :1st_place_medal:

**Reply by riongull** [reply]
Glad you found a way to add your view (I'm not focussing on permissions yet, just data structure, which permissions depend on).
Which view is yours?  Maybe you could name it "dashameter" or something?

Other than permissions, do you have any specific feature requests, bugs reports, or comments on the overall structure/organization?

**Reply by cloudwheels** [reply]
@dashameter @riongull can you set me as maintainer or whatever to be able to create a view - thanks

**Reply by riongull** [reply]
Try it now

**Comment by cloudwheels** [comment]
Is this supposed to be a full historical import of tasks or are there some sort of criteria/cut-off points?
I also seem to have duplicated trello card level imports.

**Reply by riongull** [reply]
It should have all active bounties/cards and tasks from the time of import. 

I'm mostly trying to see if the overall structure of data works for people. You should have access create views now so you can play around with the different ways to view data.

**Comment by cloudwheels** [comment]
Created https://github.com/orgs/dashtesting/projects/4/views/13?groupedBy%5BcolumnId%5D=144882
- Tried grouping by Milestone but we have none set - would it be useful to use these?
- Does grouping actually remain permenantly with the view (it seems to be in the link)
- Can't have multiple grouping levels, so although my grouping of tasks by bounty is useful, doesn't look like you could have a parent grouping by status/milsestone etc.

I'm sure there's plenty more to explore here though!

**Reply by riongull** [reply]
Yes. They would be useful, but we don't have any milestone data from Trello so that would be something to discuss after we're on GitHub. 

I'm trying to see if there's anything about the way we imported data that our admins don't feel is ideal. After that we can add incremental improvements like adding milestone data.

We could look into pulling in task due dates, but most of that would be in the past. Future due dates are somewhat loose from what I've seen (correct me if I'm wrong, and if this data would be useful).

**Comment by wizlee** [comment]
@riongull Below are my observation/feedback before creating view, I think I don't have the permission to create a view yet. [This link](https://github.com/orgs/dashtesting/projects/4/views/1?filterQuery=trello-type%3ACard&groupedBy%5BcolumnId%5D=Status) seems to represent the view that I am trying now.

What I liked:
- The ability to view everything in table and to be able to filter them.
- The `group by` feature is neat too. It is something like a pivot table and provides additional flexibility on how we filter and view the table.

Potential area of improvement:
- The `Dash Bounty` column. It will be great if we could filter it as an integer instead of string, that way we can filter by something like >1 
- For a new developer trying this table view, he/she might be quite confuse at the start if they are coming from or expecting a card based view similar to Trello. Having said that it shouldn't be a major issue as long as we refine a clean master view for everyone to land on. Maybe that is by defaulting to the board style, or put more details at the right sidebar (where the landing URL will have that sidebar expanded by default until the user manually closed it)

**Reply by riongull** [reply]
Thanks for your great feedback, @wizlee.  Check the email you use for GitHub now.  You should have an invite that will allow you to join the org as a member now, and then you'll be able to create a view. 

Check this out for an example Trello-like board.  
https://github.com/orgs/dashtesting/projects/4/views/4
We can make it almost exactly like our existing board, plus it will have the ability to filter.

Yes, we need to update the `Dash bounty` field to be an integer.  Good catch!
